app/models/draft_content_item.rb

```diff
@@ -35,6 +35,29 @@ class DraftContentItem < ActiveRecord::Base
   }
   validates_with RoutesAndRedirectsValidator
 
+  validates :description, well_formed_content_types: { must_include: "text/html" }
+  validates :details, well_formed_content_types: { must_include: "text/html" }
+
+  # Postgres's JSON columns have problems storing literal strings, so these
+  # getter/setter overrides wrap and unwrap those strings in hashes.
+  def description=(value)
+    if value.is_a?(String)
+      super(string: value)
+    else
+      super
+    end
+  end
+
+  def description
+    value = super
+
+    if value.is_a?(Hash) && value.key?(:string)
+      value.fetch(:string)
+    else
+      value
+    end
+  end
```

Comment:
My only concern with this is that it could make it harder to index this data if we ever need to search it. This may not be a problem in practice though.